content/blog/2023/08/30/2023-08-30-a-glimpse-of-the-future.adoc

```diff
@@ -0,0 +1,460 @@
+We're running the last weekly release of Jenkins that can handle JDK21, so why should we keep that? +
+We've got another fly in the ointment:
+the existing jenkins script in `/usr/bin/jenkins` is pulling a rabbit out of its hat with `sed`,
+trying to pin down a valid release.
+Unfortunately, our Java binary spits out something like “`21-beta`”.
+The script then trips up on validation, and in the end, throws a tantrum and refuses to start Jenkins. +
+Our workaround (while we twiddle our thumbs
+waiting for link:https://github.com/jenkinsci/packaging/pull/429[this PR] to build the next package) is to tweak this file.
+Change line 40 so it reads:
+
+[source,shell]
+----
+awk -F '"' '/version/ \{print $2}' | awk -F '.' '\{match($1, /^[0-9]+/); print substr($1, RSTART, RLENGTH)}')
+----
+
+Once you've done that, fire off `sudo systemctl daemon-reload` and `sudo systemctl restart jenkins` and voilà,
+you've got a Jenkins instance running on JDK21 on a RISC-V machine. +
+For those of you still hanging in there, yes, there's a quicker workaround... +
+Just stick the `Environment="JENKINS_OPTS=--enable-future-java"` line back in the `override.conf` file,
+and you'll have a Jenkins instance up and running with JDK21 faster than you can say "compile".
+
+image:/images/post-images/2023/08/30/2023-08-30-a-glimpse-of-the-future/image1.png[image,width=839]
+
+== What’s next?
+
+We've now got a fresh-off-the-press Jenkins instance humming along thanks to a nightly build of the JDK21 by Temurin. +
+That's all well and good, but how do we install newer versions of the JDK when they roll off the assembly line? +
+One of these days, we'll have regular JDK21 releases available straight from the package manager,
+but until that day comes, how are we supposed to update our installed version?
+
+Well, I don't have a silver bullet, but how about using Jenkins? +
+I've whipped up a nifty little link:https://github.com/gounthar/jenkins-temurin-riscv/blob/main/Jenkinsfile-21[Jenkinsfile] that checks every day at 2:30 AM UTC if Temurin has pushed out a new nightly build for RISC-V. +
+If it strikes gold, it installs it into `/home/jenkins/jdk-21`. +
+It's not exactly a masterpiece, since the version we're using is in `/opt/jdk21`,
+but we could tweak it to shoot us an email when it finds something. +
+We'd then have to put Jenkins on ice, shuffle the contents of `/home/jenkins/jdk-21` over to `/opt/jdk21`,
+and then wake Jenkins back up.
+
+If you're itching to give it a whirl,
+just whip up a link:/doc/book/pipeline/getting-started/#defining-a-pipeline-in-scm[new pipeline from source control]
+and point it at the link:https://github.com/gounthar/jenkins-temurin-riscv.git[repo I mentioned earlier]. +
+I'll be the first to admit, even with an email notification, this workflow isn't exactly a walk in the park,
+but it should be a stopgap solution, since Java 21 is set to launch on September 19, 2023.
```

Comment:
```suggestion
If you're itching to give it a whirl, just whip up a link:/doc/book/pipeline/getting-started/#defining-a-pipeline-in-scm[new pipeline from source control] and point it at the link:https://github.com/gounthar/jenkins-temurin-riscv.git[repo I mentioned earlier]. +
I'll be the first to admit, even with an email notification, this workflow isn't exactly a walk in the park, but it should be a stopgap solution, since Java 21 is set to launch on September 19, 2023.
```
suggestion for formatting